cluster/environments/azure-advanced/README.md

```diff
@@ -0,0 +1,32 @@
+## Getting started with azure-advanced environment
+
+The `azure-advanced` builds upon the `azure-simple` environment and adds the following funcitonality:
+
+- Provisions an [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/)
+- Configure Key Vault support within the AKS cluster making use of [Kubernetes Key Vault Flex Volume](https://github.com/Azure/kubernetes-keyvault-flexvol)
+
+To deploy the `azure-advanced` environment, follow the [common steps](../../azure/) taking into account the additional configuration required for each of the added components:
+
+- [Azure Key Vault](#azure-key-vault)
+- [Kubernetes Key Vault Flex Volume](#kubernetes-key-vault-flex-volume)
+
+Additional variables that can be configured for `azure-advanced`, see [variables.tf](./variables.tf).
+
+*NOTE*: Because the deployment of Flex Volume requires additional [Azure Role Assignments](https://docs.microsoft.com/en-us/rest/api/authorization/roleassignments) be created.  The Service Principal provisioned as part of the [common setup](../../azure/README.md#create-an-azure-service-principal) is required to have `Owner` level privileges on the subscription.
+
+### Azure Key Vault
+
+[Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/) is a secure repository for secets, keys, and certificates.  Within this environment, it is used to store and manage secrets that could be used by the AKS deployment itself or applications running within the cluster.  In order to configure Azure Key Vault in this environment, the following additional variables must be definted:
+
+- `keyvault_name`: this is a unique name for the Key Vault.  Per [this doc](https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates), Key Vault names are are globally unique and must be a 3-24 character string, containing only 0-9, a-z, A-Z, and -.
+- `secret_name`: the name of a secret to create within the Key Vault
```

Comment:
Can we add these to the terraform.tfvars in the template so folks know they need to fill them out?  Also, let's add some description of what these do - its not clear to me from the description why we need to create secrets here?

Reply:
I'll update this, is it needed?  Unknown.  But showing how to do such is useful.